pkg/configs/advisory/v2/package_advisory.go

```diff
@@ -0,0 +1,12 @@
+package v2
+
+// PackageAdvisory is an Advisory that includes the package name for that
+// advisory. (The Advisory type does not include the package's name.)
+type PackageAdvisory struct {
+	PackageName string `yaml:"packageName" json:"packageName"`
+	Advisory
+}
+
+func (pa PackageAdvisory) IsZero() bool {
```

Comment:
Can we add a comment here? Does it make sense to even have an advisory that does not have a package name? I would expect the interface to only return valid entries back if that's what it's for?

Reply:
Yeah I'll add a comment. This is following Go's `IsZero` idiom.

>Does it make sense to even have an advisory that does not have a package name? I would expect the interface to only return valid entries back if that's what it's for?

It doesn't make sense, and I agree, I would expect implementations of the interface to only return usable data. This `IsZero` method isn't on the interface but on the `PackageAdvisory` type, so there's no inherent guarantee that we have data populated, and this method is useful for data validation.

Reply:
Updated in 4a01ebb

Reply:
Maybe add to interface definition that the values returned in the array for `Advisories` is guaranteed to be 'non-zero'.
And yeah, since it's on the type I get that, but I'm looking at the return type from the interface, and I'm like do I need to call this for returned types. So in that sense while it's struct, because it's returned from the interface was curious.

Reply:
Ah I get it now. Yes, I'll make that part of the interface expectations so no one has to check.